File: backend/ride_track/ride_track_app/application/user/get_user_by_id.py
from django.http.response import JsonResponse
from rest_framework import status

from ride_track_app.domain.user.model import User
from ride_track_app.domain.user.serializer import UserSerializer
from .response import UserResponse

def get_user_by_id(id):
    user_query = User.objects.filter(id=id)
    if user_query.count() == 0:
        error = {
            'error': 'user not found'
        }
        return JsonResponse(error, status=status.HTTP_404_NOT_FOUND)

    user_serializer = UserSerializer(user_query[0])
    user_response = UserResponse(
            user_serializer.data.get('first_name'),
            user_serializer.data.get('last_name'),
            user_serializer.data.get('email'),
            user_serializer.data.get('id'))
    return JsonResponse(user_response.__dict__, status=status.HTTP_200_OK)

File: backend/ride-track-nodejs/src/application/user/get-user-by-id.ts
import { IUserRepository } from "../../domain/user/iuser-repository";
import { TYPES } from "../../inversify-types";
import { inject } from "inversify";
import { container } from "../../inversify-config";
import { ApplicationResponse, Status } from "../application-response";
import { Error } from "../error";
import { UserResponse } from "../../contract/user/user-response";

export class GetUserByIdQuery {
    @inject(TYPES.IUserRepository) private userRepository: IUserRepository 
        = container.get<IUserRepository>(TYPES.IUserRepository);
    
    public getUserById = async (id: string): Promise<ApplicationResponse> => {
        const user = await this.userRepository.getUserById(id);
        if (user == undefined) {
            return {
                status: Status.NotFound,
                resource: new Error('User not found')
            };
        }

        return {
            status: Status.Ok,
            resource: new UserResponse(
                user.first_name, 
                user.last_name, 
                user.weight,
                user.email,
                user.id,
                user.unit)
        };
    }
}